### ISSUE TITLE
Type hints not working for `qtpy` specific imports

### ISSUE THREAD

**Issue by wkrasnicki** [spyder-ide/qtpy#447]
Hi,

I want to start an incremental migration from PyQt5 using your amazing project. However, I immediately hit an obstacle, when renaming my imports. The problem is with the abstracted `Signal`, `Slot` etc. classes. Their type hints are not picked up by VSCode's intellisense, which is a bit of a deal breaker for me :/

I looked into your `QtCore` file and it seems that this is caused by exporting Signal as both `pyqtSignal` from PyQt5 and `Unknown` from other packages. When I comment out other imports, I get the correct type, but I'm not familiar enough with Python typing hacks to figure out how to solve it. Is it even possible? I would very much like this feature and could submit a PR if you have some advice what to try.

**Comment by dalthviz** [comment]
Hi @wkrasnicki thank you for the feedback! Not totally sure how to proceed here (my knowledge about Python typings checking is quite limited to be honest) but which are the imports you are commenting out to enable the type checking to work? Maybe we just need to add a `if not TYPE_CHECKING` validation to such lines to ensure the typing works but also still have the actual imports?

Any other info is greatly appreciated! Let us know!

**Comment by wkrasnicki** [comment]
Hello, I have produced a minimal example of the issue. It requires creating 3 files and assumes only PyQt5 is installed.

```
# env.py
PYQT5 = 1 == 1
PYQT6 = 1 == 2

# qtcore.py
from .env import PYQT6, PYQT5

if PYQT5:
    from PyQt5.QtCore import pyqtSignal as Signal
elif PYQT6:
    from PyQt6.QtCore import pyqtSignal as Signal

# main.py
from .qtcore import Signal

Signal()
```

When launched in VSCode with basic type checking, I expect this to show no red lines from type checking. There is a minor difference in the actual `qtpy`, where the variables from `env.py` are of type `Literal[False]` and `Literal[True]` respectively, but changing them to bool did not help, so here I am also using bools. I'm trying out different things, but no progress so far.

**Comment by dalthviz** [comment]
Thank you for the info @wkrasnicki ! I checked and I'm not seeing any red lines 🤔 Just in case, in your actual usage of QtPy do you have the config for mypy and QtPy set on VSCode? For more info you can check: https://github.com/spyder-ide/qtpy/blob/master/README.md#type-checker-integration (more specifically the link to the Pyright config comment).

Let us now if the info above helps!

**Comment by dalthviz** [comment]
Awesome @wkrasnicki glad to know you manage to config things in your setup! Regarding the README, for sure there are quite a few opportunities for improvement :) Do you think giving the Pyright info its own title/section could be better, right? For sure at least updating the info regarding the current way to config Pyright  needs to be done. If you want to help us with that let us know!

**Comment by wkrasnicki** [comment]
Hi @dalthviz I've created a PR with proposed changes. Let me know what you think :)

**Comment by ccordoba12** [comment]
Closing as fixed by #450.
